diff -ur ray.orig/src/cal/calc/Makefile ray/src/cal/calc/Makefile
--- ray.orig/src/cal/calc/Makefile	Sat Jul  6 18:20:18 2002
+++ ray/src/cal/calc/Makefile	Sat Jul  6 22:31:06 2002
@@ -24,7 +24,7 @@
 caldefn.o  calfunc.o calprnt.o ealloc.o savestr.o $(COMPAT) $(MLIB)
 
 install:	calc
-	cp calc $(INSTDIR)
+	install calc $(DESTDIR)$(INSTDIR)
 
 clean:
 	set nonomatch; rm -f *.o calc
diff -ur ray.orig/src/cal/calc/Rmakefile ray/src/cal/calc/Rmakefile
--- ray.orig/src/cal/calc/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/cal/calc/Rmakefile	Sat Jul  6 22:31:06 2002
@@ -24,7 +24,7 @@
 caldefn.o  calfunc.o calprnt.o ealloc.o savestr.o $(COMPAT) $(MLIB)
 
 install:	calc
-	cp calc $(INSTDIR)
+	install calc $(DESTDIR)$(INSTDIR)
 
 clean:
 	set nonomatch; rm -f *.o calc
diff -ur ray.orig/src/cal/ev/Makefile ray/src/cal/ev/Makefile
--- ray.orig/src/cal/ev/Makefile	Sat Jul  6 18:20:18 2002
+++ ray/src/cal/ev/Makefile	Sat Jul  6 22:31:06 2002
@@ -21,7 +21,7 @@
 calfunc.o ealloc.o savestr.o $(MLIB)
 
 install:	ev
-	cp ev $(INSTDIR)
+	install ev $(DESTDIR)$(INSTDIR)
 
 clean:
 	set nonomatch; rm -f *.o ev
diff -ur ray.orig/src/cal/ev/Rmakefile ray/src/cal/ev/Rmakefile
--- ray.orig/src/cal/ev/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/cal/ev/Rmakefile	Sat Jul  6 22:31:06 2002
@@ -21,7 +21,7 @@
 calfunc.o ealloc.o savestr.o $(MLIB)
 
 install:	ev
-	cp ev $(INSTDIR)
+	install ev $(DESTDIR)$(INSTDIR)
 
 clean:
 	set nonomatch; rm -f *.o ev
diff -ur ray.orig/src/cal/rcalc/Makefile ray/src/cal/rcalc/Makefile
--- ray.orig/src/cal/rcalc/Makefile	Sat Jul  6 18:20:18 2002
+++ ray/src/cal/rcalc/Makefile	Sat Jul  6 22:31:06 2002
@@ -24,7 +24,7 @@
 caldefn.o  calfunc.o ealloc.o savestr.o $(COMPAT) $(MLIB)
 
 install:	rcalc
-	cp rcalc $(INSTDIR)
+	install rcalc $(DESTDIR)$(INSTDIR)
 
 clean:
 	set nonomatch; rm -f *.o rcalc
diff -ur ray.orig/src/cal/rcalc/Rmakefile ray/src/cal/rcalc/Rmakefile
--- ray.orig/src/cal/rcalc/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/cal/rcalc/Rmakefile	Sat Jul  6 22:31:06 2002
@@ -24,7 +24,7 @@
 caldefn.o  calfunc.o ealloc.o savestr.o $(COMPAT) $(MLIB)
 
 install:	rcalc
-	cp rcalc $(INSTDIR)
+	install rcalc $(DESTDIR)$(INSTDIR)
 
 clean:
 	set nonomatch; rm -f *.o rcalc
diff -ur ray.orig/src/cal/util/Makefile ray/src/cal/util/Makefile
--- ray.orig/src/cal/util/Makefile	Sat Jul  6 18:20:18 2002
+++ ray/src/cal/util/Makefile	Sat Jul  6 22:31:06 2002
@@ -19,7 +19,7 @@
 all:	$(PROGS)
 
 install:	$(PROGS)
-	cp $(PROGS) $(INSTDIR)
+	install $(PROGS) $(DESTDIR)$(INSTDIR)
 
 clean:
 	set nonomatch; rm -f *.o $(PROGS)
diff -ur ray.orig/src/cal/util/Rmakefile ray/src/cal/util/Rmakefile
--- ray.orig/src/cal/util/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/cal/util/Rmakefile	Sat Jul  6 22:31:06 2002
@@ -19,7 +19,7 @@
 all:	$(PROGS)
 
 install:	$(PROGS)
-	cp $(PROGS) $(INSTDIR)
+	install $(PROGS) $(DESTDIR)$(INSTDIR)
 
 clean:
 	set nonomatch; rm -f *.o $(PROGS)
diff -ur ray.orig/src/common/Rmakefile ray/src/common/Rmakefile
--- ray.orig/src/common/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/common/Rmakefile	Sat Jul  6 22:31:06 2002
@@ -31,17 +31,18 @@
 
 OGLOBJ = rglfile.o rglmat.o rgldomat.o rglsurf.o rglinst.o rglsrc.o
 
-all:	librt.a $(SPECIAL)
+all:	librayt.a $(SPECIAL)
 
-librt.a:	$(RTOBJ) $(PICOBJ) $(STDOBJ) $(SYSOBJ)
-	ar rc librt.a $(RTOBJ) $(PICOBJ) $(STDOBJ) $(SYSOBJ)
-	-ranlib librt.a
+librayt.a:	$(RTOBJ) $(PICOBJ) $(STDOBJ) $(SYSOBJ)
+	ar rc librayt.a $(RTOBJ) $(PICOBJ) $(STDOBJ) $(SYSOBJ)
+	-ranlib librayt.a
+	cp librayt.a ../lib/librayt.a
 
 install:	all
-	mv librt.a ../lib/librt.a
+	mv librayt.a ../lib/librayt.a
 
 clean:
-	rm -f *.o librt.a
+	rm -f *.o librayt.a
 
 ogl:	$(OGLOBJ)
 	ar rc librgl.a $(OGLOBJ)
diff -ur ray.orig/src/common/lookup.h ray/src/common/lookup.h
--- ray.orig/src/common/lookup.h	Sat Jul  6 18:20:18 2002
+++ ray/src/common/lookup.h	Sat Jul  6 22:31:06 2002
@@ -5,6 +5,7 @@
 /*
  * Header file for general associative table lookup routines
  */
+#include <string.h>
 
 typedef struct {
 	char	*key;			/* key name */
@@ -80,4 +81,3 @@
 extern void	lu_done();
 extern unsigned long	lu_shash();
 
-extern int	strcmp();
diff -ur ray.orig/src/cv/Rmakefile ray/src/cv/Rmakefile
--- ray.orig/src/cv/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/cv/Rmakefile	Sat Jul  6 22:31:06 2002
@@ -22,12 +22,14 @@
 
 LIBFILES = source.cal tilt.cal lamp.tab tmesh.cal
 
+MGF_LIB = ../lib/libmgf.a
+
 all:	$(PROGS)
 
 install:	$(PROGS) $(LIBFILES)
-	cp $(PROGS) $(INSTDIR)
-	cd $(LIBDIR) ; rm -f $(LIBFILES)
-	cp $(LIBFILES) $(LIBDIR)
+	install $(PROGS) $(DESTDIR)$(INSTDIR)
+	cd $(DESTDIR)$(LIBDIR) ; rm -f $(LIBFILES)
+	install $(LIBFILES) $(DESTDIR)$(LIBDIR)
 
 clean:
 	set nonomatch; rm -f $(PROGS) *.o core
@@ -37,31 +39,31 @@
 	$(CC) $(CFLAGS) -o thf2rad thf2rad.o
 
 ies2rad:	ies2rad.o
-	$(CC) $(CFLAGS) -o ies2rad ies2rad.o -lrt -lm
+	$(CC) $(CFLAGS) -o ies2rad ies2rad.o -lrayt -lm
 
 arch2rad:	arch2rad.o trans.o
-	$(CC) $(CFLAGS) -o arch2rad arch2rad.o trans.o -lrt
+	$(CC) $(CFLAGS) -o arch2rad arch2rad.o trans.o -lrayt
 
 nff2rad:	nff2rad.o
 	$(CC) $(CFLAGS) -o nff2rad nff2rad.o
 
 lampcolor:	lampcolor.o
-	$(CC) $(CFLAGS) -o lampcolor lampcolor.o -lrt
+	$(CC) $(CFLAGS) -o lampcolor lampcolor.o -lrayt
 
 tmesh2rad:	tmesh2rad.o tmesh.o
-	$(CC) $(CFLAGS) -o tmesh2rad tmesh2rad.o tmesh.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o tmesh2rad tmesh2rad.o tmesh.o -lrayt $(MLIB)
 
 obj2rad:	obj2rad.o trans.o tmesh.o
-	$(CC) $(CFLAGS) -o obj2rad obj2rad.o trans.o tmesh.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o obj2rad obj2rad.o trans.o tmesh.o -lrayt $(MLIB)
 
-mgf2rad:	MGF_LIB mgf2rad.o tmesh.o
-	$(CC) $(CFLAGS) -o mgf2rad mgf2rad.o tmesh.o -lmgf -lrt $(MLIB)
+mgf2rad:	$(MGF_LIB) mgf2rad.o tmesh.o
+	$(CC) $(CFLAGS) -o mgf2rad mgf2rad.o tmesh.o -lmgf -lrayt $(MLIB)
 
 rad2mgf:	rad2mgf.o
-	$(CC) $(CFLAGS) -o rad2mgf rad2mgf.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o rad2mgf rad2mgf.o -lrayt $(MLIB)
 
-mgf2meta:	MGF_LIB mgf2meta.o
-	$(CC) $(CFLAGS) -o mgf2meta mgf2meta.o -lmgf -lmeta -lrt $(MLIB)
+mgf2meta:	$(MGF_LIB) mgf2meta.o
+	$(CC) $(CFLAGS) -o mgf2meta mgf2meta.o -lmgf -lmeta -lrayt $(MLIB)
 
 mgfilt:
 	cd mgflib ; rm -f mgfilt ; \
@@ -75,7 +77,7 @@
 		CFLAGS="$(OPT) $(MACH) -DNOPROTO" ; \
 	mv mgf2inv ..
 
-MGF_LIB:
+$(MGF_LIB):
 	cd mgflib ; rm -f libmgf.a ; \
 	make libmgf.a CC=$(CC) \
 		CFLAGS="$(OPT) $(MACH) -DNOPROTO" ; \
diff -ur ray.orig/src/gen/Rmakefile ray/src/gen/Rmakefile
--- ray.orig/src/gen/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/gen/Rmakefile	Sat Jul  6 22:31:06 2002
@@ -22,9 +22,9 @@
 all:	$(PROGS)
 
 install:	$(PROGS) $(LIBFILES)
-	cp $(PROGS) $(INSTDIR)
-	cd $(LIBDIR) ; rm -f $(LIBFILES)
-	cp $(LIBFILES) $(LIBDIR)
+	install $(PROGS) $(DESTDIR)$(INSTDIR)
+	cd $(DESTDIR)$(LIBDIR) ; rm -f $(LIBFILES)
+	install $(LIBFILES) $(DESTDIR)$(LIBDIR)
 
 clean:
 	set nonomatch; rm -f $(PROGS) *.o core
@@ -48,22 +48,22 @@
 	$(CC) $(CFLAGS) -o genprism genprism.o $(MLIB)
 
 xform:	xform.o
-	$(CC) $(CFLAGS) -o xform xform.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o xform xform.o -lrayt $(MLIB)
 
 replmarks:	replmarks.o
-	$(CC) $(CFLAGS) -o replmarks replmarks.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o replmarks replmarks.o -lrayt $(MLIB)
 
 mkillum:	mkillum.o mkillum2.o mkillum3.o
-	$(CC) $(CFLAGS) -o mkillum mkillum.o mkillum2.o mkillum3.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o mkillum mkillum.o mkillum2.o mkillum3.o -lrayt $(MLIB)
 
 genrev:	genrev.o 
-	$(CC) $(CFLAGS) -o genrev genrev.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o genrev genrev.o -lrayt $(MLIB)
 
 genworm:	genworm.o
-	$(CC) $(CFLAGS) -o genworm genworm.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o genworm genworm.o -lrayt $(MLIB)
 
 gensurf:	gensurf.o
-	$(CC) $(CFLAGS) -o gensurf gensurf.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o gensurf gensurf.o -lrayt $(MLIB)
 
 genclock:	genclock.o
 	$(CC) $(CFLAGS) -o genclock genclock.o $(MLIB)
diff -ur ray.orig/src/hd/Rmakefile ray/src/hd/Rmakefile
--- ray.orig/src/hd/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/hd/Rmakefile	Sat Jul  6 22:46:00 2002
@@ -14,31 +14,30 @@
 SPECIAL =
 
 INSTDIR = /usr/local/bin
-DEVDIR = $(INSTDIR)/dev
 
 LIBDIR = /usr/local/lib/ray
 
 PROGS = rholo rhpict rhcopy genrhgrid rhoptimize rhinfo
 
-DEVS = $(DEVDIR)/x11.hdi $(DEVDIR)/x11h.hdi
+DEVS = x11.hdi x11h.hdi
 
 all:	$(PROGS)
 
 install:	all $(DEVS) $(SPECIAL)
-	cp $(PROGS) $(INSTDIR)
+	install $(PROGS) $(DESTDIR)$(INSTDIR)
 
 clean:
 	set nonomatch; rm -f *.o $(PROGS) core
 
-ogl:	$(DEVDIR)/glx.hdi $(DEVDIR)/glxh.hdi \
-	$(DEVDIR)/glxo.hdi $(DEVDIR)/glxoh.hdi \
-	$(DEVDIR)/glxs.hdi $(DEVDIR)/glxsh.hdi \
-	$(DEVDIR)/glxso.hdi $(DEVDIR)/glxsoh.hdi \
-	$(DEVDIR)/glx1.hdi $(DEVDIR)/glx1h.hdi \
-	$(DEVDIR)/ogl.hdi $(DEVDIR)/oglh.hdi \
-	$(DEVDIR)/oglo.hdi $(DEVDIR)/ogloh.hdi \
-	$(DEVDIR)/ogls.hdi $(DEVDIR)/oglsh.hdi \
-	$(DEVDIR)/oglso.hdi $(DEVDIR)/oglsoh.hdi
+ogl:	glx.hdi glxh.hdi \
+	glxo.hdi glxoh.hdi \
+	glxs.hdi glxsh.hdi \
+	glxso.hdi glxsoh.hdi \
+	glx1.hdi glx1h.hdi \
+	ogl.hdi oglh.hdi \
+	oglo.hdi ogloh.hdi \
+	ogls.hdi oglsh.hdi \
+	oglso.hdi oglsoh.hdi
 
 x10:
 
@@ -51,135 +50,135 @@
 rholo:	rholo.o rholo2.o rholo2l.o rholo3.o rholo4.o holo.o holofile.o \
 viewbeams.o malloc.o Version.o
 	$(CC) $(CFLAGS) -o rholo rholo.o rholo2.o rholo2l.o rholo3.o rholo4.o \
-holo.o holofile.o viewbeams.o malloc.o Version.o -lrt -lm
+holo.o holofile.o viewbeams.o malloc.o Version.o -lrayt -lm
 
 rhpict:	rhpict.o rhpict2.o holo.o holofile.o viewbeams.o malloc.o Version.o
 	$(CC) $(CFLAGS) -o rhpict rhpict.o rhpict2.o holo.o holofile.o \
-viewbeams.o malloc.o Version.o -lrt -lm
+viewbeams.o malloc.o Version.o -lrayt -lm
 
 rhcopy:	rhcopy.o clumpbeams.o holo.o holofile.o
 	$(CC) $(CFLAGS) -o rhcopy rhcopy.o clumpbeams.o \
-holo.o holofile.o -lrt -lm
+holo.o holofile.o -lrayt -lm
 
 rhoptimize:	rhoptimize.o clumpbeams.o holo.o holofile.o
 	$(CC) $(CFLAGS) -o rhoptimize rhoptimize.o clumpbeams.o \
-holo.o holofile.o -lrt -lm
+holo.o holofile.o -lrayt -lm
 
 rhinfo:	rhinfo.o holo.o holofile.o
-	$(CC) $(CFLAGS) -o rhinfo rhinfo.o holo.o holofile.o -lrt -lm
+	$(CC) $(CFLAGS) -o rhinfo rhinfo.o holo.o holofile.o -lrayt -lm
 
 genrhenv:	genrhenv.o holo.o holofile.o
-	$(CC) $(CFLAGS) -o genrhenv genrhenv.o holo.o holofile.o -lrt -lm
+	$(CC) $(CFLAGS) -o genrhenv genrhenv.o holo.o holofile.o -lrayt -lm
 
 genrhgrid:	genrhgrid.o holo.o
-	$(CC) $(CFLAGS) -o genrhgrid genrhgrid.o holo.o -lrt -lm
+	$(CC) $(CFLAGS) -o genrhgrid genrhgrid.o holo.o -lrayt -lm
 
-$(DEVDIR)/x11h.hdi:	$(DEVDIR)/x11.hdi
-	rm -f $(DEVDIR)/x11h.hdi
-	ln $(DEVDIR)/x11.hdi $(DEVDIR)/x11h.hdi
+x11h.hdi:	x11.hdi
+	rm -f x11h.hdi
+	ln x11.hdi x11h.hdi
 
-$(DEVDIR)/x11.hdi:	rhdisp.o rhdisp2.o viewbeams.o rhd_x11.o rhd_qtree.o \
+x11.hdi:	rhdisp.o rhdisp2.o viewbeams.o rhd_x11.o rhd_qtree.o \
 rhd_qtree2r.o rhd_ctab.o dircode.o holo.o
-	$(CC) $(CFLAGS) -o $(DEVDIR)/x11.hdi rhdisp.o rhdisp2.o viewbeams.o \
-rhd_x11.o rhd_qtree.o rhd_qtree2r.o rhd_ctab.o holo.o dircode.o -lrt -lX11 -lm
+	$(CC) $(CFLAGS) -o x11.hdi rhdisp.o rhdisp2.o viewbeams.o \
+rhd_x11.o rhd_qtree.o rhd_qtree2r.o rhd_ctab.o holo.o dircode.o -lrayt -lX11 -lm -L/usr/X11R6/lib
 
-$(DEVDIR)/glx1h.hdi:	$(DEVDIR)/glx1.hdi
-	rm -f $(DEVDIR)/glx1h.hdi
-	ln $(DEVDIR)/glx1.hdi $(DEVDIR)/glx1h.hdi
-
-$(DEVDIR)/glxh.hdi:	$(DEVDIR)/glx.hdi
-	rm -f $(DEVDIR)/glxh.hdi
-	ln $(DEVDIR)/glx.hdi $(DEVDIR)/glxh.hdi
-
-$(DEVDIR)/glxoh.hdi:	$(DEVDIR)/glxo.hdi
-	rm -f $(DEVDIR)/glxoh.hdi
-	ln $(DEVDIR)/glxo.hdi $(DEVDIR)/glxoh.hdi
-
-$(DEVDIR)/glxsh.hdi:	$(DEVDIR)/glxs.hdi
-	rm -f $(DEVDIR)/glxsh.hdi
-	ln $(DEVDIR)/glxs.hdi $(DEVDIR)/glxsh.hdi
-
-$(DEVDIR)/glxsoh.hdi:	$(DEVDIR)/glxso.hdi
-	rm -f $(DEVDIR)/glxsoh.hdi
-	ln $(DEVDIR)/glxso.hdi $(DEVDIR)/glxsoh.hdi
-
-$(DEVDIR)/oglh.hdi:	$(DEVDIR)/ogl.hdi
-	rm -f $(DEVDIR)/oglh.hdi
-	ln $(DEVDIR)/ogl.hdi $(DEVDIR)/oglh.hdi
-
-$(DEVDIR)/ogloh.hdi:	$(DEVDIR)/oglo.hdi
-	rm -f $(DEVDIR)/ogloh.hdi
-	ln $(DEVDIR)/oglo.hdi $(DEVDIR)/ogloh.hdi
-
-$(DEVDIR)/oglsh.hdi:	$(DEVDIR)/ogls.hdi
-	rm -f $(DEVDIR)/oglsh.hdi
-	ln $(DEVDIR)/ogls.hdi $(DEVDIR)/oglsh.hdi
-
-$(DEVDIR)/oglsoh.hdi:	$(DEVDIR)/oglso.hdi
-	rm -f $(DEVDIR)/oglsoh.hdi
-	ln $(DEVDIR)/oglso.hdi $(DEVDIR)/oglsoh.hdi
+glx1h.hdi:	glx1.hdi
+	rm -f glx1h.hdi
+	ln glx1.hdi glx1h.hdi
+
+glxh.hdi:	glx.hdi
+	rm -f glxh.hdi
+	ln glx.hdi glxh.hdi
+
+glxoh.hdi:	glxo.hdi
+	rm -f glxoh.hdi
+	ln glxo.hdi glxoh.hdi
+
+glxsh.hdi:	glxs.hdi
+	rm -f glxsh.hdi
+	ln glxs.hdi glxsh.hdi
+
+glxsoh.hdi:	glxso.hdi
+	rm -f glxsoh.hdi
+	ln glxso.hdi glxsoh.hdi
+
+oglh.hdi:	ogl.hdi
+	rm -f oglh.hdi
+	ln ogl.hdi oglh.hdi
+
+ogloh.hdi:	oglo.hdi
+	rm -f ogloh.hdi
+	ln oglo.hdi ogloh.hdi
+
+oglsh.hdi:	ogls.hdi
+	rm -f oglsh.hdi
+	ln ogls.hdi oglsh.hdi
+
+oglsoh.hdi:	oglso.hdi
+	rm -f oglsoh.hdi
+	ln oglso.hdi oglsoh.hdi
 
-$(DEVDIR)/glx1.hdi:	rhd_glx1.o rhdisp.o rhdisp2.o viewbeams.o \
+glx1.hdi:	rhd_glx1.o rhdisp.o rhdisp2.o viewbeams.o \
 rhd_qtree.o rhd_qtree2c.o dircode.o holo.o
-	rm -f $(DEVDIR)/glx1.hdi
-	$(CC) $(CFLAGS) -o $(DEVDIR)/glx1.hdi rhd_glx1.o rhdisp.o rhdisp2.o \
-viewbeams.o holo.o malloc.o rhd_qtree.o rhd_qtree2c.o dircode.o -lrt -lGLU -lGL -lX11 -lm
+	rm -f glx1.hdi
+	$(CC) $(CFLAGS) -o glx1.hdi rhd_glx1.o rhdisp.o rhdisp2.o \
+viewbeams.o holo.o malloc.o rhd_qtree.o rhd_qtree2c.o dircode.o -lrayt -lGLU -lGL -lX11 -lm
 
-$(DEVDIR)/glx.hdi:	rhd_glx0.o rhdisp.o rhdisp2.o viewbeams.o \
+glx.hdi:	rhd_glx0.o rhdisp.o rhdisp2.o viewbeams.o \
 sm_samp.o sm_qtree.o sm_stree.o sm_geom.o sm_list.o sm_del.o \
 sm_ogl.o sm.o sm_usets.o dircode.o holo.o
-	rm -f $(DEVDIR)/glx.hdi
-	$(CC) $(CFLAGS) -o $(DEVDIR)/glx.hdi rhd_glx0.o rhdisp.o rhdisp2.o \
+	rm -f glx.hdi
+	$(CC) $(CFLAGS) -o glx.hdi rhd_glx0.o rhdisp.o rhdisp2.o \
 viewbeams.o holo.o sm_samp.o sm_qtree.o sm_stree.o sm_geom.o sm_list.o sm_del.o \
-sm_ogl.o sm.o sm_usets.o dircode.o -lrt -lGLU -lGL -lX11 -lm
+sm_ogl.o sm.o sm_usets.o dircode.o -lrayt -lGLU -lGL -lX11 -lm
 
-$(DEVDIR)/glxo.hdi:	rhd_glxo.o rhdisp.o rhdisp2.o viewbeams.o \
+glxo.hdi:	rhd_glxo.o rhdisp.o rhdisp2.o viewbeams.o \
 sm_samp.o sm_qtree.o sm_stree.o sm_geom.o sm_list.o sm_del.o \
 sm_ogl.o sm_usets.o sm.o dircode.o holo.o rhdobj.o 
-	rm -f $(DEVDIR)/glxo.hdi
-	$(CC) $(CFLAGS) -o $(DEVDIR)/glxo.hdi rhd_glxo.o rhdisp.o rhdisp2.o \
+	rm -f glxo.hdi
+	$(CC) $(CFLAGS) -o glxo.hdi rhd_glxo.o rhdisp.o rhdisp2.o \
 viewbeams.o holo.o sm_samp.o sm_qtree.o sm_stree.o sm_geom.o sm_list.o sm_del.o \
-sm_ogl.o sm_usets.o sm.o dircode.o rhdobj.o -lrgl -lrt -lGLU -lGL -lX11 -lm
+sm_ogl.o sm_usets.o sm.o dircode.o rhdobj.o -lrgl -lrayt -lGLU -lGL -lX11 -lm
 
-$(DEVDIR)/glxs.hdi:	rhd_glxs.o rhdisp.o rhdisp2.o viewbeams.o \
+glxs.hdi:	rhd_glxs.o rhdisp.o rhdisp2.o viewbeams.o \
 sm_samp.o sm_qtree.o sm_stree.o sm_geom.o sm_list.o sm_del.o \
 sm_ogl.o sm_usets.o sm.o dircode.o holo.o
-	rm -f $(DEVDIR)/glxs.hdi
-	$(CC) $(CFLAGS) -o $(DEVDIR)/glxs.hdi rhd_glxs.o rhdisp.o rhdisp2.o \
+	rm -f glxs.hdi
+	$(CC) $(CFLAGS) -o glxs.hdi rhd_glxs.o rhdisp.o rhdisp2.o \
 viewbeams.o holo.o sm_samp.o sm_qtree.o sm_stree.o sm_geom.o sm_list.o sm_del.o \
-sm_ogl.o sm_usets.o sm.o dircode.o -lrt -lGLU -lGL -lX11 -lXext -lm
+sm_ogl.o sm_usets.o sm.o dircode.o -lrayt -lGLU -lGL -lX11 -lXext -lm
 
-$(DEVDIR)/glxso.hdi:	rhd_glxso.o rhdisp.o rhdisp2.o viewbeams.o \
+glxso.hdi:	rhd_glxso.o rhdisp.o rhdisp2.o viewbeams.o \
 sm_samp.o sm_qtree.o sm_stree.o sm_geom.o sm_list.o sm_del.o \
 sm_ogl.o sm_usets.o sm.o dircode.o holo.o rhdobj.o
-	rm -f $(DEVDIR)/glxso.hdi
-	$(CC) $(CFLAGS) -o $(DEVDIR)/glxso.hdi rhd_glxso.o rhdisp.o rhdisp2.o \
+	rm -f glxso.hdi
+	$(CC) $(CFLAGS) -o glxso.hdi rhd_glxso.o rhdisp.o rhdisp2.o \
 viewbeams.o holo.o sm_samp.o sm_qtree.o sm_stree.o sm_geom.o sm_list.o sm_del.o \
 sm_ogl.o sm_usets.o sm.o dircode.o rhdobj.o \
--lrgl -lrt -lGLU -lGL -lX11 -lXext -lm
+-lrgl -lrayt -lGLU -lGL -lX11 -lXext -lm
 
-$(DEVDIR)/ogl.hdi:	rhd_ogl0.o rhdisp.o rhdisp2.o viewbeams.o \
+ogl.hdi:	rhd_ogl0.o rhdisp.o rhdisp2.o viewbeams.o \
 rhd_odraw.o rhd_geom.o holo.o
-	$(CC) $(CFLAGS) -o $(DEVDIR)/ogl.hdi rhd_ogl0.o rhdisp.o rhdisp2.o \
-viewbeams.o holo.o rhd_odraw.o rhd_geom.o -lrgl -lrt -lGLU -lGL -lX11 -lm
+	$(CC) $(CFLAGS) -o ogl.hdi rhd_ogl0.o rhdisp.o rhdisp2.o \
+viewbeams.o holo.o rhd_odraw.o rhd_geom.o -lrgl -lrayt -lGLU -lGL -lX11 -lm
 
-$(DEVDIR)/oglo.hdi:	rhd_oglo.o rhdisp.o rhdisp2.o viewbeams.o \
+oglo.hdi:	rhd_oglo.o rhdisp.o rhdisp2.o viewbeams.o \
 rhd_odraw.o rhd_geom.o holo.o rhdobj.o
-	$(CC) $(CFLAGS) -o $(DEVDIR)/oglo.hdi rhd_oglo.o rhdisp.o \
+	$(CC) $(CFLAGS) -o oglo.hdi rhd_oglo.o rhdisp.o \
 rhdisp2.o viewbeams.o holo.o rhd_odraw.o rhd_geom.o rhdobj.o \
--lrgl -lrt -lGLU -lGL -lX11 -lm
+-lrgl -lrayt -lGLU -lGL -lX11 -lm
 
-$(DEVDIR)/ogls.hdi:	rhd_ogls.o rhdisp.o rhdisp2.o viewbeams.o \
+ogls.hdi:	rhd_ogls.o rhdisp.o rhdisp2.o viewbeams.o \
 rhd_odraw.o rhd_geom.o holo.o
-	$(CC) $(CFLAGS) -o $(DEVDIR)/ogls.hdi rhd_ogls.o rhdisp.o rhdisp2.o \
+	$(CC) $(CFLAGS) -o ogls.hdi rhd_ogls.o rhdisp.o rhdisp2.o \
 viewbeams.o holo.o rhd_odraw.o rhd_geom.o \
--lrgl -lrt -lGLU -lGL -lX11 -lXext -lm
+-lrgl -lrayt -lGLU -lGL -lX11 -lXext -lm
 
-$(DEVDIR)/oglso.hdi:	rhd_oglso.o rhdisp.o rhdisp2.o viewbeams.o \
+oglso.hdi:	rhd_oglso.o rhdisp.o rhdisp2.o viewbeams.o \
 rhd_odraw.o rhd_geom.o holo.o rhdobj.o
-	$(CC) $(CFLAGS) -o $(DEVDIR)/oglso.hdi rhd_oglso.o rhdisp.o rhdisp2.o \
+	$(CC) $(CFLAGS) -o oglso.hdi rhd_oglso.o rhdisp.o rhdisp2.o \
 viewbeams.o holo.o rhd_odraw.o rhd_geom.o rhdobj.o \
--lrgl -lrt -lGLU -lGL -lX11 -lXext -lm
+-lrgl -lrayt -lGLU -lGL -lX11 -lXext -lm
 
 rhd_glx0.o:	rhd_glx.c
 	$(CC) $(CFLAGS) -c rhd_glx.c
diff -ur ray.orig/src/meta/Rmakefile ray/src/meta/Rmakefile
--- ray.orig/src/meta/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/meta/Rmakefile	Sat Jul  6 22:31:06 2002
@@ -24,127 +24,98 @@
 
 common = $(io) syscalls.o misc.o savestr.o wputs.o eputs.o quit.o $(COMPAT)
 
-install:	general
+PROGS = mx80 meta2tga t4014 pexpand psort cv psmeta plotin imagew impress x11meta \
+	bgraph igraph dgraph gcomp aed5 plot4
+PROGS2 = mt160l mt160 mtext plotout xmeta okimate tbar
+
+all:  $(common) ../lib/libmeta.a $(PROGS)
+
+install:	$(PROGS)
+	install $(PROGS) $(DESTDIR)$(INSTDIR)
 
 clean:
 	set nonomatch; rm -f *.o a.out core
 
-general:	$(INSTDIR)/mx80 $(INSTDIR)/meta2tga \
-$(INSTDIR)/t4014 $(INSTDIR)/pexpand $(INSTDIR)/psort \
-$(INSTDIR)/cv $(INSTDIR)/psmeta $(EXEC2)/plotin \
-$(INSTDIR)/imagew $(INSTDIR)/impress $(INSTDIR)/x11meta \
-../lib/libmeta.a $(INSTDIR)/bgraph $(INSTDIR)/igraph \
-$(INSTDIR)/dgraph $(INSTDIR)/gcomp $(INSTDIR)/aed5 $(INSTDIR)/plot4
-
-special:	$(EXEC2)/mt160l $(EXEC2)/mt160 $(EXEC2)/mtext \
-$(EXEC2)/plotout $(EXEC2)/xmeta $(EXEC2)/okimate $(EXEC2)/tbar
-
-$(INSTDIR)/meta2tga $(EXEC2)/okimate $(INSTDIR)/impress \
-$(INSTDIR)/mx80 $(INSTDIR)/t4014 $(INSTDIR)/plotout $(INSTDIR)/x11meta \
-$(EXEC2)/mt160l $(EXEC2)/mt160 $(INSTDIR)/pexpand $(INSTDIR)/plot4 \
-$(INSTDIR)/psort $(INSTDIR)/imagew $(INSTDIR)/tcurve $(EXEC2)/mtext \
-$(INSTDIR)/tscat $(INSTDIR)/plotin $(INSTDIR)/tbar $(INSTDIR)/xmeta \
-../lib/libmeta.a $(INSTDIR)/aed5 $(INSTDIR)/psmeta:	$(common)
-
-$(EXEC2)/mt160l:	mt160l.o mplot.o plot.o palloc.o
-	$(CC) $(CFLAGS) -o $(EXEC2)/mt160l mt160l.o mplot.o plot.o \
+mt160l:	mt160l.o mplot.o plot.o palloc.o
+	$(CC) $(CFLAGS) -o mt160l mt160l.o mplot.o plot.o \
 $(common) palloc.o
 
-$(EXEC2)/okimate:	okimate.o mplot.o plot.o palloc.o
-	$(CC) $(CFLAGS) -o $(EXEC2)/okimate okimate.o mplot.o plot.o \
+okimate:	okimate.o mplot.o plot.o palloc.o
+	$(CC) $(CFLAGS) -o okimate okimate.o mplot.o plot.o \
 $(common) palloc.o
 
-$(INSTDIR)/imagew:	imagew.o mplot.o plot.o palloc.o
+imagew:	imagew.o mplot.o plot.o palloc.o
 	$(CC) $(CFLAGS) imagew.o mplot.o plot.o $(common) \
-palloc.o $(MLIB)
-	cp a.out $(INSTDIR)/imagew
+palloc.o $(MLIB) -o imagew
 
-$(EXEC2)/mt160:	mt160.o mplot.o plot.o palloc.o
-	$(CC) $(CFLAGS) -o $(EXEC2)/mt160 mt160.o mplot.o plot.o $(common) \
+mt160:	mt160.o mplot.o plot.o palloc.o
+	$(CC) $(CFLAGS) -o mt160 mt160.o mplot.o plot.o $(common) \
 palloc.o
 
-$(INSTDIR)/meta2tga:	meta2tga.o rplot.o plot.o palloc.o
+meta2tga:	meta2tga.o rplot.o plot.o palloc.o
 	$(CC) $(CFLAGS) meta2tga.o rplot.o plot.o $(common) \
-palloc.o $(MLIB)
-	cp a.out $(INSTDIR)/meta2tga
+palloc.o $(MLIB) -o meta2tga
 
-$(INSTDIR)/mx80:	mx80.o mplot.o plot.o palloc.o
+mx80:	mx80.o mplot.o plot.o palloc.o
 	$(CC) $(CFLAGS) mx80.o mplot.o plot.o $(common) \
-palloc.o $(MLIB)
-	cp a.out $(INSTDIR)/mx80
+palloc.o $(MLIB) -o mx80
 
-$(INSTDIR)/t4014:	plotout.c ../lib/lib4014.a
+t4014:	plotout.c ../lib/lib4014.a
 	$(CC) -DFORTEK $(CFLAGS) plotout.c $(common) \
-../lib/lib4014.a $(MLIB)
-	cp a.out $(INSTDIR)/t4014
+../lib/lib4014.a $(MLIB) -o t4014
 	rm -f plotout.o
 
-$(INSTDIR)/impress:	impress.o implot.o imPfuncs.o plot.o
+impress:	impress.o implot.o imPfuncs.o plot.o
 	$(CC) $(CFLAGS) impress.o implot.o imPfuncs.o \
-plot.o $(common) $(MLIB)
-	cp a.out $(INSTDIR)/impress
+plot.o $(common) $(MLIB) -o impress
+
+aed5:	aed5.o
+	$(CC) $(CFLAGS) aed5.o $(common) $(MLIB) -o aed5
 
-$(INSTDIR)/aed5:	aed5.o
-	$(CC) $(CFLAGS) aed5.o $(common) $(MLIB)
-	cp a.out $(INSTDIR)/aed5
-
-$(INSTDIR)/xmeta:	xmeta.o xplot.o plot.o palloc.o
-	$(CC) $(CFLAGS) xmeta.o xplot.o plot.o palloc.o $(common) -lX11 $(MLIB)
-	cp a.out $(INSTDIR)/xmeta
-
-$(INSTDIR)/x11meta:	xmeta.o x11plot.o plot.o palloc.o
-	$(CC) $(CFLAGS) xmeta.o x11plot.o plot.o palloc.o $(common) -lX11 $(MLIB)
-	cp a.out $(INSTDIR)/x11meta
-
-$(INSTDIR)/psmeta:	psmeta.o psplot.o
-	$(CC) $(CFLAGS) psmeta.o psplot.o $(common) $(MLIB)
-	cp a.out $(INSTDIR)/psmeta
-
-$(INSTDIR)/plotout:	plotout.c
-	$(CC) $(CFLAGS) plotout.c $(common) -lplot $(MLIB)
-	cp a.out $(INSTDIR)/plotout
+xmeta:	xmeta.o xplot.o plot.o palloc.o
+	$(CC) $(CFLAGS) xmeta.o xplot.o plot.o palloc.o $(common) -lX11 $(MLIB) -o xmeta
+
+x11meta:	xmeta.o x11plot.o plot.o palloc.o
+	$(CC) $(CFLAGS) xmeta.o x11plot.o plot.o palloc.o $(common) -lX11 $(MLIB) -o x11meta
+
+psmeta:	psmeta.o psplot.o
+	$(CC) $(CFLAGS) psmeta.o psplot.o $(common) $(MLIB) -o psmeta
+
+plotout:	plotout.c
+	$(CC) $(CFLAGS) plotout.c $(common) -lplot $(MLIB) -o plotout
 	rm -f plotout.o
 
-$(INSTDIR)/plotin:	plotin.o primout.o
-	$(CC) $(CFLAGS) plotin.o primout.o $(common) $(MLIB)
-	cp a.out $(INSTDIR)/plotin
+plotin:	plotin.o primout.o
+	$(CC) $(CFLAGS) plotin.o primout.o $(common) $(MLIB) -o plotin
 
-$(INSTDIR)/pexpand:	pexpand.o expand.o segment.o palloc.o
+pexpand:	pexpand.o expand.o segment.o palloc.o
 	$(CC) $(CFLAGS) pexpand.o expand.o segment.o \
-$(common) palloc.o $(MLIB)
-	cp a.out $(INSTDIR)/pexpand
+$(common) palloc.o $(MLIB) -o pexpand
 
-$(INSTDIR)/psort:	psort.o sort.o palloc.o
-	$(CC) $(CFLAGS) psort.o sort.o $(common) palloc.o $(MLIB)
-	cp a.out $(INSTDIR)/psort
+psort:	psort.o sort.o palloc.o
+	$(CC) $(CFLAGS) psort.o sort.o $(common) palloc.o $(MLIB) -o psort
 
-$(INSTDIR)/cv:		cv.o mfio.o cvhfio.o syscalls.o misc.o savestr.o \
+cv:		cv.o mfio.o cvhfio.o syscalls.o misc.o savestr.o \
 wputs.o eputs.o quit.o $(COMPAT)
 	$(CC) $(CFLAGS) cv.o mfio.o cvhfio.o syscalls.o misc.o \
-savestr.o wputs.o eputs.o quit.o $(COMPAT) $(MLIB)
-	cp a.out $(INSTDIR)/cv
+savestr.o wputs.o eputs.o quit.o $(COMPAT) $(MLIB) -o cv
 
-$(INSTDIR)/plot4:	plot4.o primout.o
-	$(CC) $(CFLAGS) plot4.o primout.o $(common) $(MLIB)
-	cp a.out $(INSTDIR)/plot4
+plot4:	plot4.o primout.o
+	$(CC) $(CFLAGS) plot4.o primout.o $(common) $(MLIB) -o plot4
 
-$(INSTDIR)/tcurve:	tcurve.o tgraph.o primout.o
+tcurve:	tcurve.o tgraph.o primout.o
 	$(CC) $(CFLAGS) tcurve.o tgraph.o primout.o \
-$(common) $(MLIB)
-	cp a.out $(INSTDIR)/tcurve
+$(common) $(MLIB) -o tcurve
 
-$(INSTDIR)/tscat:	tscat.o tgraph.o primout.o
+tscat:	tscat.o tgraph.o primout.o
 	$(CC) $(CFLAGS) tscat.o tgraph.o primout.o \
-$(common) $(MLIB)
-	cp a.out $(INSTDIR)/tscat
+$(common) $(MLIB) -o tscat
+
+tbar:	tbar.o tgraph.o primout.o
+	$(CC) $(CFLAGS) tbar.o tgraph.o primout.o $(common) $(MLIB) -o tbar
 
-$(INSTDIR)/tbar:	tbar.o tgraph.o primout.o
-	$(CC) $(CFLAGS) tbar.o tgraph.o primout.o $(common) $(MLIB)
-	cp a.out $(INSTDIR)/tbar
-
-$(EXEC2)/mtext:	mtext.o primout.o
-	$(CC) $(CFLAGS) mtext.o primout.o $(common)
-	cp a.out $(EXEC2)/mtext
+mtext:	mtext.o primout.o
+	$(CC) $(CFLAGS) mtext.o primout.o $(common) -o mtext
 
 ../lib/libmeta.a:	metacalls.o primout.o progname.o
 	ar rc ../lib/libmeta.a metacalls.o primout.o \
@@ -154,30 +125,26 @@
 ../lib/lib4014.a:
 	cd lib4014; make INSTDIR=../../lib "CFLAGS=$(OPT)" install clean
 
-$(INSTDIR)/bgraph:	bgraph.o mgvars.o mgraph.o ../lib/libmeta.a \
+bgraph:	bgraph.o mgvars.o mgraph.o ../lib/libmeta.a \
 calexpr.o caldefn.o calfunc.o ealloc.o
 	$(CC) $(CFLAGS) bgraph.o mgvars.o mgraph.o \
-calexpr.o caldefn.o calfunc.o ealloc.o ../lib/libmeta.a $(MLIB)
-	cp a.out $(INSTDIR)/bgraph
+calexpr.o caldefn.o calfunc.o ealloc.o ../lib/libmeta.a $(MLIB) -o bgraph
 
-$(INSTDIR)/igraph:	igraph.o mgvars.o mgraph.o ../lib/libmeta.a \
+igraph:	igraph.o mgvars.o mgraph.o ../lib/libmeta.a \
 calexpr.o caldefn.o calfunc.o cgraph.o ealloc.o gcalc.o
 	$(CC) $(CFLAGS) igraph.o mgvars.o mgraph.o \
 calexpr.o caldefn.o calfunc.o cgraph.o ealloc.o gcalc.o \
-../lib/libmeta.a $(MLIB)
-	cp a.out $(INSTDIR)/igraph
+../lib/libmeta.a $(MLIB) -o igraph
 
-$(INSTDIR)/dgraph:	dgraph.o cgraph.o mgvars.o ealloc.o calfunc.o \
+dgraph:	dgraph.o cgraph.o mgvars.o ealloc.o calfunc.o \
 calexpr.o caldefn.o savestr.o wputs.o $(COMPAT)
 	$(CC) $(CFLAGS) dgraph.o cgraph.o mgvars.o \
-calfunc.o calexpr.o caldefn.o ealloc.o savestr.o wputs.o $(COMPAT) $(MLIB)
-	cp a.out $(INSTDIR)/dgraph
+calfunc.o calexpr.o caldefn.o ealloc.o savestr.o wputs.o $(COMPAT) $(MLIB) -o dgraph
 
-$(INSTDIR)/gcomp:	gcomp.o gcalc.o mgvars.o ealloc.o calfunc.o \
+gcomp:	gcomp.o gcalc.o mgvars.o ealloc.o calfunc.o \
 calexpr.o caldefn.o wputs.o savestr.o $(COMPAT)
 	$(CC) $(CFLAGS) gcomp.o gcalc.o mgvars.o \
-calfunc.o calexpr.o caldefn.o ealloc.o wputs.o savestr.o $(COMPAT) $(MLIB)
-	cp a.out $(INSTDIR)/gcomp
+calfunc.o calexpr.o caldefn.o ealloc.o wputs.o savestr.o $(COMPAT) $(MLIB) -o gcomp
 
 calexpr.o:	calexpr.c calcomp.h
 	$(CC) $(CFLAGS) -c -DVARIABLE -DFUNCTION -DRCONST calexpr.c
diff -ur ray.orig/src/meta/igraph.c ray/src/meta/igraph.c
--- ray.orig/src/meta/igraph.c	Sat Jul  6 18:20:18 2002
+++ ray/src/meta/igraph.c	Sat Jul  6 22:31:06 2002
@@ -14,6 +14,8 @@
 
 #include  "mgvars.h"
 
+#include "meta.h"
+
 typedef struct {
 	char  *descrip;		/* description */
 	char  *value;		/* value */
diff -ur ray.orig/src/meta/meta.h ray/src/meta/meta.h
--- ray.orig/src/meta/meta.h	Sat Jul  6 18:20:18 2002
+++ ray/src/meta/meta.h	Sat Jul  6 22:31:06 2002
@@ -90,6 +90,7 @@
 #define  YMN  1			/* index in xy array for ymin */
 #define  XMX  2			/* index in xy array for xmax */
 #define  YMX  3			/* index in xy array for ymax */
+#define pout stdout
 
 
 /*
diff -ur ray.orig/src/meta/meta2tga.c ray/src/meta/meta2tga.c
--- ray.orig/src/meta/meta2tga.c	Sat Jul  6 18:20:18 2002
+++ ray/src/meta/meta2tga.c	Sat Jul  6 22:31:06 2002
@@ -67,6 +67,8 @@
 #endif
  char  comargs[200], command[300];
 
+ fout=stdout;
+
 #ifdef  CPM
  fixargs("meta2tga", &argc, &argv);
 #endif
diff -ur ray.orig/src/ot/Rmakefile ray/src/ot/Rmakefile
--- ray.orig/src/ot/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/ot/Rmakefile	Sat Jul  6 22:31:06 2002
@@ -24,14 +24,14 @@
 o_cone.o o_instance.o bbox.o initotypes.o readfargs.o malloc.o
 	$(CC) $(CFLAGS) -o oconv oconv.o writeoct.o sphere.o o_face.o \
 o_cone.o o_instance.o bbox.o readfargs.o \
-initotypes.o malloc.o -lrt $(MLIB)
+initotypes.o malloc.o -lrayt $(MLIB)
 
 getbbox:	getbbox.o readobj2.o bbox.o init2otypes.o
 	$(CC) $(CFLAGS) -o getbbox getbbox.o readobj2.o \
-bbox.o init2otypes.o -lrt $(MLIB)
+bbox.o init2otypes.o -lrayt $(MLIB)
 
 install:	oconv getbbox
-	cp oconv getbbox $(INSTDIR)
+	install oconv getbbox $(DESTDIR)$(INSTDIR)
 
 clean:
 	set nonomatch; rm -f oconv getbbox *.o core
diff -ur ray.orig/src/px/Rmakefile ray/src/px/Rmakefile
--- ray.orig/src/px/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/px/Rmakefile	Sat Jul  6 22:31:06 2002
@@ -5,7 +5,7 @@
 ARCH = sun
 OPT = -O
 MACH = -DBSD
-SPECIAL = aed
+# SPECIAL = aed
 CFLAGS = -I../common -L../lib $(OPT) $(MACH)
 CC = cc
 MLIB = -lm
@@ -25,13 +25,13 @@
 all:	$(PROGS) $(SPECIAL)
 
 x10:	x10image
-	cp x10image $(INSTDIR)
+	cp x10image $(DESTDIR)$(INSTDIR)
 
 aed:	aedimage
-	cp aedimage $(INSTDIR)
+	cp aedimage $(DESTDIR)$(INSTDIR)
 
 sgi:	glimage
-	cp glimage $(INSTDIR)
+	cp glimage $(DESTDIR)$(INSTDIR)
 
 ogl:
 
@@ -39,15 +39,15 @@
 
 install:	$(PROGS) $(SPECIAL) normpat.csh falsecolor.csh \
 pdfblur.csh pmblur.csh xyzimage.csh
-	cp $(PROGS) $(INSTDIR)
-	cp normpat.csh $(INSTDIR)/normpat
-	cp falsecolor.csh $(INSTDIR)/falsecolor
-	cp pdfblur.csh $(INSTDIR)/pdfblur
-	cp pmblur.csh $(INSTDIR)/pmblur
-	cp xyzimage.csh $(INSTDIR)/xyzimage
-	cp phisto.csh $(INSTDIR)/phisto
-	chmod 755 $(INSTDIR)/normpat $(INSTDIR)/falsecolor \
-$(INSTDIR)/pdfblur $(INSTDIR)/pmblur $(INSTDIR)/xyzimage $(INSTDIR)/phisto
+	install $(PROGS) $(DESTDIR)$(INSTDIR)
+	install normpat.csh $(DESTDIR)$(INSTDIR)/normpat
+	install falsecolor.csh $(DESTDIR)$(INSTDIR)/falsecolor
+	install pdfblur.csh $(DESTDIR)$(INSTDIR)/pdfblur
+	install pmblur.csh $(DESTDIR)$(INSTDIR)/pmblur
+	install xyzimage.csh $(DESTDIR)$(INSTDIR)/xyzimage
+	install phisto.csh $(DESTDIR)$(INSTDIR)/phisto
+	chmod 755 $(DESTDIR)$(INSTDIR)/normpat $(DESTDIR)$(INSTDIR)/falsecolor \
+$(DESTDIR)$(INSTDIR)/pdfblur $(DESTDIR)$(INSTDIR)/pmblur $(DESTDIR)$(INSTDIR)/xyzimage $(DESTDIR)$(INSTDIR)/phisto
 
 clean:
 	set nonomatch; rm -f x10image aedimage glimage \
@@ -55,140 +55,140 @@
 	cd tiff; make distclean
 
 pfilt:	pfilt.o pf2.o pf3.o
-	$(CC) $(CFLAGS) -o pfilt pfilt.o pf2.o pf3.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o pfilt pfilt.o pf2.o pf3.o -lrayt $(MLIB)
 
 pcond:	pcond.o pcond2.o pcond3.o pcond4.o warp3d.o
 	$(CC) $(CFLAGS) -o pcond pcond.o pcond2.o pcond3.o pcond4.o warp3d.o \
--lrt $(MLIB)
+-lrayt $(MLIB)
 
 pcwarp:	pcwarp.o warp3d.o
-	$(CC) $(CFLAGS) -o pcwarp pcwarp.o warp3d.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o pcwarp pcwarp.o warp3d.o -lrayt $(MLIB)
 
 ttyimage:	ttyimage.o
-	$(CC) $(CFLAGS) -o ttyimage ttyimage.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o ttyimage ttyimage.o -lrayt $(MLIB)
 
 t4027:	t4027.o
-	$(CC) $(CFLAGS) -o t4027 t4027.o -lrt
+	$(CC) $(CFLAGS) -o t4027 t4027.o -lrayt
 
 glimage:	glimage.o
-	$(CC) $(CFLAGS) -o glimage glimage.o -lrt -lgl_s $(MLIB)
+	$(CC) $(CFLAGS) -o glimage glimage.o -lrayt -lgl_s $(MLIB)
 
 aedimage:	aedimage.o ciq.o cut.o closest.o biq.o
 	$(CC) $(CFLAGS) -o aedimage aedimage.o ciq.o cut.o closest.o \
-biq.o -lrt $(MLIB)
+biq.o -lrayt $(MLIB)
 
 oki20c:	oki20c.o
-	$(CC) $(CFLAGS) -o oki20c oki20c.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o oki20c oki20c.o -lrayt $(MLIB)
 
 oki20:	oki20.o
-	$(CC) $(CFLAGS) -o oki20 oki20.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o oki20 oki20.o -lrayt $(MLIB)
 
 paintjet:	paintjet.o
-	$(CC) $(CFLAGS) -o paintjet paintjet.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o paintjet paintjet.o -lrayt $(MLIB)
 
 mt160r:	mt160r.o
-	$(CC) $(CFLAGS) -o mt160r mt160r.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o mt160r mt160r.o -lrayt $(MLIB)
 
 greyscale:	greyscale.o
-	$(CC) $(CFLAGS) -o greyscale greyscale.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o greyscale greyscale.o -lrayt $(MLIB)
 
 colorscale:	colorscale.o
-	$(CC) $(CFLAGS) -o colorscale colorscale.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o colorscale colorscale.o -lrayt $(MLIB)
 
 pvalue:	pvalue.o
-	$(CC) $(CFLAGS) -o pvalue pvalue.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o pvalue pvalue.o -lrayt $(MLIB)
 
 psum:	psum.o
-	$(CC) $(CFLAGS) -o psum psum.o -lrt
+	$(CC) $(CFLAGS) -o psum psum.o -lrayt
 
 pcompos:	pcompos.o
-	$(CC) $(CFLAGS) -o pcompos pcompos.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o pcompos pcompos.o -lrayt $(MLIB)
 
 psign:	psign.o
-	$(CC) $(CFLAGS) -o psign psign.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o psign psign.o -lrayt $(MLIB)
 
 d48c:	d48c.o
-	$(CC) $(CFLAGS) -o d48c d48c.o -lrt
+	$(CC) $(CFLAGS) -o d48c d48c.o -lrayt
 
 ra_gif:	ra_gif.o clrtab.o neuclrtab.o
-	$(CC) $(CFLAGS) -o ra_gif ra_gif.o clrtab.o neuclrtab.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o ra_gif ra_gif.o clrtab.o neuclrtab.o -lrayt $(MLIB)
 
 ra_pr:	ra_pr.o ciq.o cut.o closest.o biq.o
 	$(CC) $(CFLAGS) -o ra_pr ra_pr.o ciq.o cut.o closest.o \
-biq.o -lrt $(MLIB)
+biq.o -lrayt $(MLIB)
 
 ra_pr24:	ra_pr24.o
-	$(CC) $(CFLAGS) -o ra_pr24 ra_pr24.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o ra_pr24 ra_pr24.o -lrayt $(MLIB)
 
 ra_avs:	ra_avs.o
-	$(CC) $(CFLAGS) -o ra_avs ra_avs.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o ra_avs ra_avs.o -lrayt $(MLIB)
 
 ra_ps:	ra_ps.o
-	$(CC) $(CFLAGS) -o ra_ps ra_ps.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o ra_ps ra_ps.o -lrayt $(MLIB)
 
 ra_ppm:		ra_ppm.o
-	$(CC) $(CFLAGS) -o ra_ppm ra_ppm.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o ra_ppm ra_ppm.o -lrayt $(MLIB)
 
 ra_t8:	ra_t8.o clrtab.o neuclrtab.o
-	$(CC) $(CFLAGS) -o ra_t8 ra_t8.o clrtab.o neuclrtab.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o ra_t8 ra_t8.o clrtab.o neuclrtab.o -lrayt $(MLIB)
 
 ra_t16: ra_t16.o
-	$(CC) $(CFLAGS) -o ra_t16 ra_t16.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o ra_t16 ra_t16.o -lrayt $(MLIB)
 
 ra_im: ra_im.o
 	$(CC) $(CFLAGS) -o ra_im ra_im.o
 
 ra_bn: ra_bn.o
-	$(CC) $(CFLAGS) -o ra_bn ra_bn.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o ra_bn ra_bn.o -lrayt $(MLIB)
 
 ra_rgbe:	ra_rgbe.o
-	$(CC) $(CFLAGS) -o ra_rgbe ra_rgbe.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o ra_rgbe ra_rgbe.o -lrayt $(MLIB)
 
 ra_pict:    ra_pict.o
-	$(CC) $(CFLAGS) -o ra_pict ra_pict.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o ra_pict ra_pict.o -lrayt $(MLIB)
 
 ra_hexbit:	ra_hexbit.o
-	$(CC) $(CFLAGS) -o ra_hexbit ra_hexbit.o -lrt
+	$(CC) $(CFLAGS) -o ra_hexbit ra_hexbit.o -lrayt
 
 x10image:	ximage.o ciq.o cut.o closest.o xraster.o biq.o
 	$(CC) $(CFLAGS) -o x10image ximage.o xraster.o \
-ciq.o cut.o closest.o biq.o -lX -lrt $(MLIB)
+ciq.o cut.o closest.o biq.o -lX -lrayt $(MLIB)
 
 ximage:	x11image.o x11raster.o clrtab.o
 	$(CC) $(CFLAGS) -o ximage x11image.o x11raster.o \
-clrtab.o -lrt -lX11 $(MLIB)
+clrtab.o -lrayt -lX11 $(MLIB)
 
 protate:		protate.o
-	$(CC) $(CFLAGS) -o protate protate.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o protate protate.o -lrayt $(MLIB)
 
 pextrem:	pextrem.o
-	$(CC) $(CFLAGS) -o pextrem pextrem.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o pextrem pextrem.o -lrayt $(MLIB)
 
 pflip:		pflip.o
-	$(CC) $(CFLAGS) -o pflip pflip.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o pflip pflip.o -lrayt $(MLIB)
 
 panim:	panim.o client/libclient.a
 	$(CC) $(CFLAGS) -o panim panim.o \
--lrt client/libclient.a -lrpcsvc $(MLIB)
+-lrayt client/libclient.a -lrpcsvc $(MLIB)
 
 pcomb:	pcomb.o
-	$(CC) $(CFLAGS) -o pcomb pcomb.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o pcomb pcomb.o -lrayt $(MLIB)
 
 pinterp:	pinterp.o
-	$(CC) $(CFLAGS) -o pinterp pinterp.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o pinterp pinterp.o -lrayt $(MLIB)
 
 ra_xyze:	ra_xyze.o
-	$(CC) $(CFLAGS) -o ra_xyze ra_xyze.o -lrt $(MLIB)
+	$(CC) $(CFLAGS) -o ra_xyze ra_xyze.o -lrayt $(MLIB)
 
 ra_tiff:	ra_tiff.o ../lib/libtiff.a
-	$(CC) $(CFLAGS) -o ra_tiff ra_tiff.o -lrt -ltiff $(MLIB)
+	$(CC) $(CFLAGS) -o ra_tiff ra_tiff.o -lrayt -ltiff $(MLIB)
 
 ../common/tiff.h ../common/tiffio.h:	../lib/libtiff.a
 	@chmod u+w ../common/tiff.h ../common/tiffio.h
 	@touch ../common/tiff.h ../common/tiffio.h
 
 normtiff:	normtiff.o ../lib/libtiff.a
-	$(CC) $(CFLAGS) -o normtiff normtiff.o -lrt -ltiff $(MLIB)
+	$(CC) $(CFLAGS) -o normtiff normtiff.o -lrayt -ltiff $(MLIB)
 
 ../lib/libtiff.a:	tiff/config.local
 	cd tiff ; ./configure -quiet -noninteractive -with-CC=$(CC) \
@@ -196,7 +196,7 @@
 	cd libtiff ; make install
 
 ra_pixar:	ra_pixar.o ra_pixar.c
-	$(CC) $(CFLAGS) $(PIXAR_INCLUDE) -o $@ ra_pixar.o -lrt \
+	$(CC) $(CFLAGS) $(PIXAR_INCLUDE) -o $@ ra_pixar.o -lrayt \
 $(PIXAR_LIB)
 
 ra_pixar.o:	ra_pixar.c
@@ -204,11 +204,11 @@
 
 xshowtrace:	xshowtrace.o x11findwind.o
 	$(CC) $(CFLAGS) -o xshowtrace xshowtrace.o \
-x11findwind.o -lrt -lX11 $(MLIB)
+x11findwind.o -lrayt -lX11 $(MLIB)
 
 macbethcal:	macbethcal.o pmapgen.o mx3.o warp3d.o
 	$(CC) $(CFLAGS) -o macbethcal macbethcal.o pmapgen.o mx3.o warp3d.o \
--lrt $(MLIB)
+-lrayt $(MLIB)
 
 
 macbethcal.o:	pmap.h ../common/color.h ../common/resolu.h
diff -ur ray.orig/src/px/pcwarp.c ray/src/px/pcwarp.c
--- ray.orig/src/px/pcwarp.c	Sat Jul  6 18:20:18 2002
+++ ray/src/px/pcwarp.c	Sat Jul  6 22:31:06 2002
@@ -30,6 +30,7 @@
 	int	cwflags = 0;
 	int	rval;
 	int	i;
+	infp=stdin;
 
 	progname = argv[0];
 	infp = stdin;
diff -ur ray.orig/src/px/x11image.c ray/src/px/x11image.c
--- ray.orig/src/px/x11image.c	Sat Jul  6 18:20:18 2002
+++ ray/src/px/x11image.c	Sat Jul  6 22:31:06 2002
@@ -141,6 +141,7 @@
 	int  headline();
 	int  i;
 	int  pid;
+	fin=stdin;
 	
 	progname = argv[0];
 	fin = stdin;
diff -ur ray.orig/src/rt/Rmakefile ray/src/rt/Rmakefile
--- ray.orig/src/rt/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/rt/Rmakefile	Sat Jul  6 22:31:06 2002
@@ -5,8 +5,8 @@
 
 OPT = -O
 MACH = -DBSD
-CFLAGS = -I../common -L../lib $(OPT) $(MACH)
-SPECIAL = aed
+CFLAGS = -I../common -L../lib $(OPT) $(MACH) -I/usr/include/ncurses
+#SPECIAL = aed
 CC = cc
 MLIB = -lm
 LINT = lint
@@ -18,7 +18,7 @@
 #
 DESTDIR = .
 INSTDIR = /usr/local/bin
-INSTALL = cp
+INSTALL = install
 
 #
 # The following paths must exist and be relative to root:
@@ -29,7 +29,7 @@
 #
 # Library routines:
 #
-LIBS = -lrt $(MLIB)
+LIBS = -lrayt $(MLIB)
 
 #
 # Device drivers for rview (see also devtable.c):
@@ -85,14 +85,14 @@
 # What this makefile produces:
 #
 
-PROGS = $(DESTDIR)/rtrace $(DESTDIR)/rpict $(DESTDIR)/rview $(DESTDIR)/lookamb
+PROGS = rtrace rpict rview lookamb
 
 all:	$(PROGS)
 
 install:	$(PROGS) $(SPECIAL) rayinit.cal
-	$(INSTALL) $(PROGS) $(INSTDIR)
-	rm -f $(LIBDIR)/rayinit.cal
-	cp rayinit.cal $(LIBDIR)
+	$(INSTALL) $(PROGS) $(DESTDIR)$(INSTDIR)
+	rm -f $(DESTDIR)$(LIBDIR)/rayinit.cal
+	$(INSTALL) rayinit.cal $(DESTDIR)$(LIBDIR)
 
 sun:	$(DEVDIR)/sun $(DEVDIR)/sun.com
 
@@ -114,17 +114,17 @@
 # Links:
 #
 
-$(DESTDIR)/rtrace:	$(RTOBJS)
-	$(CC) $(CFLAGS) -o $(DESTDIR)/rtrace $(RTOBJS) $(LIBS)
+rtrace:	$(RTOBJS)
+	$(CC) $(CFLAGS) -o rtrace $(RTOBJS) $(LIBS)
 
-$(DESTDIR)/rpict:	$(RPOBJS)
-	$(CC) $(CFLAGS) -o $(DESTDIR)/rpict $(RPOBJS) $(LIBS)
+rpict:	$(RPOBJS)
+	$(CC) $(CFLAGS) -o rpict $(RPOBJS) $(LIBS)
 
-$(DESTDIR)/rview:	$(RVOBJS)
-	$(CC) $(CFLAGS) -o $(DESTDIR)/rview $(RVOBJS) $(LIBS) $(DLIBS)
+rview:	$(RVOBJS)
+	$(CC) $(CFLAGS) -o rview $(RVOBJS) $(LIBS) $(DLIBS)
 
-$(DESTDIR)/lookamb:	lookamb.o ambio.o
-	$(CC) $(CFLAGS) -o $(DESTDIR)/lookamb lookamb.o ambio.o $(LIBS)
+lookamb:	lookamb.o ambio.o
+	$(CC) $(CFLAGS) -o lookamb lookamb.o ambio.o $(LIBS)
 
 $(DEVDIR)/sun:	sundev.o devmain.o colortab.o editline.o
 	$(CC) $(CFLAGS) -s -o $(DEVDIR)/sun devmain.o sundev.o colortab.o \
diff -ur ray.orig/src/util/Rmakefile ray/src/util/Rmakefile
--- ray.orig/src/util/Rmakefile	Sat Jul  6 18:20:18 2002
+++ ray/src/util/Rmakefile	Sat Jul  6 22:31:06 2002
@@ -24,27 +24,27 @@
 
 install:	all $(SPECIAL) objview.csh objpict.csh glare.csh dayfact.csh \
 rlux.csh raddepend.csh trad.wsh objline.csh compamb.csh
-	cp $(PROGS) $(INSTDIR)
-	cp objview.csh $(INSTDIR)/objview
-	cp objline.csh $(INSTDIR)/objline
-	cp objpict.csh $(INSTDIR)/objpict
-	cp glare.csh $(INSTDIR)/glare
-	cp dayfact.csh $(INSTDIR)/dayfact
-	cp debugcal.csh $(INSTDIR)/debugcal
-	cp rlux.csh $(INSTDIR)/rlux
-	cp raddepend.csh $(INSTDIR)/raddepend
-	cp compamb.csh $(INSTDIR)/compamb
-	chmod 755 $(INSTDIR)/objview $(INSTDIR)/objpict \
-		$(INSTDIR)/glare $(INSTDIR)/dayfact $(INSTDIR)/debugcal \
-		$(INSTDIR)/rlux $(INSTDIR)/raddepend $(INSTDIR)/objline \
-		$(INSTDIR)/compamb
-	csh -f tradinstall.csh $(INSTDIR) $(LIBDIR)/tcl
+	cp $(PROGS) $(DESTDIR)$(INSTDIR)
+	cp objview.csh $(DESTDIR)$(INSTDIR)/objview
+	cp objline.csh $(DESTDIR)$(INSTDIR)/objline
+	cp objpict.csh $(DESTDIR)$(INSTDIR)/objpict
+	cp glare.csh $(DESTDIR)$(INSTDIR)/glare
+	cp dayfact.csh $(DESTDIR)$(INSTDIR)/dayfact
+	cp debugcal.csh $(DESTDIR)$(INSTDIR)/debugcal
+	cp rlux.csh $(DESTDIR)$(INSTDIR)/rlux
+	cp raddepend.csh $(DESTDIR)$(INSTDIR)/raddepend
+	cp compamb.csh $(DESTDIR)$(INSTDIR)/compamb
+	chmod 755 $(DESTDIR)$(INSTDIR)/objview $(DESTDIR)$(INSTDIR)/objpict \
+		$(DESTDIR)$(INSTDIR)/glare $(DESTDIR)$(INSTDIR)/dayfact $(DESTDIR)$(INSTDIR)/debugcal \
+		$(DESTDIR)$(INSTDIR)/rlux $(DESTDIR)$(INSTDIR)/raddepend $(DESTDIR)$(INSTDIR)/objline \
+		$(DESTDIR)$(INSTDIR)/compamb
+	csh -f tradinstall.csh $(DESTDIR)$(INSTDIR) $(DESTDIR)$(LIBDIR)/tcl
 
 clean:
 	set nonomatch; rm -f *.o $(PROGS) core
 
 ogl:	glrad
-	cp glrad $(INSTDIR)
+	cp glrad $(DESTDIR)$(INSTDIR)
 
 x10:
 
@@ -58,39 +58,39 @@
 	$(CC) $(CFLAGS) -o scanner scanner.o -lm
 
 makedist:	makedist.o setscan.o
-	$(CC) $(CFLAGS) -o makedist makedist.o setscan.o -lrt -lm
+	$(CC) $(CFLAGS) -o makedist makedist.o setscan.o -lrayt -lm
 
 findglare:	findglare.o glareval.o glaresrc.o setscan.o
 	$(CC) $(CFLAGS) -o findglare findglare.o glareval.o glaresrc.o \
-setscan.o -lrt -lm
+setscan.o -lrayt -lm
 
 glarendx:	glarendx.o
-	$(CC) $(CFLAGS) -o glarendx glarendx.o -lrt -lm
+	$(CC) $(CFLAGS) -o glarendx glarendx.o -lrayt -lm
 
 xglaresrc:	xglaresrc.o x11findwind.o
 	$(CC) $(CFLAGS) -o xglaresrc xglaresrc.o \
-x11findwind.o -lrt -lX11 -lm
+x11findwind.o -lrayt -lX11 -lm
 
 rad:	rad.o
-	$(CC) $(CFLAGS) -o rad rad.o -lrt -lm
+	$(CC) $(CFLAGS) -o rad rad.o -lrayt -lm
 
 ranimate:	ranimate.o netproc.o
-	$(CC) $(CFLAGS) -o ranimate ranimate.o netproc.o -lrt -lm
+	$(CC) $(CFLAGS) -o ranimate ranimate.o netproc.o -lrayt -lm
 
 rpiece:	rpiece.o Version.o
-	$(CC) $(CFLAGS) -o rpiece rpiece.o Version.o -lrt -lm
+	$(CC) $(CFLAGS) -o rpiece rpiece.o Version.o -lrayt -lm
 
 vwright:	vwright.o
-	$(CC) $(CFLAGS) -o vwright vwright.o -lrt -lm
+	$(CC) $(CFLAGS) -o vwright vwright.o -lrayt -lm
 
 vwrays:		vwrays.o
-	$(CC) $(CFLAGS) -o vwrays vwrays.o -lrt -lm
+	$(CC) $(CFLAGS) -o vwrays vwrays.o -lrayt -lm
 
 getinfo:	getinfo.o
-	$(CC) $(CFLAGS) -o getinfo getinfo.o -lrt
+	$(CC) $(CFLAGS) -o getinfo getinfo.o -lrayt
 
 glrad:	glrad.o
-	$(CC) $(CFLAGS) -o glrad glrad.o -lrgl -lrt -lGLU -lGL \
+	$(CC) $(CFLAGS) -o glrad glrad.o -lrgl -lrayt -lGLU -lGL \
 -lX11 -lXext -lm
 
 contour:	contour.o
