--- ray/src/px/Rmakefile.orig	Wed Aug 21 22:55:11 2002
+++ ray/src/px/Rmakefile	Wed Aug 21 23:09:24 2002
@@ -180,21 +180,12 @@
 ra_xyze:	ra_xyze.o
 	$(CC) $(CFLAGS) -o ra_xyze ra_xyze.o -lrayt $(MLIB)
 
-ra_tiff:	ra_tiff.o ../lib/libtiff.a
+ra_tiff:	ra_tiff.o
 	$(CC) $(CFLAGS) -o ra_tiff ra_tiff.o -lrayt -ltiff $(MLIB)
 
-../common/tiff.h ../common/tiffio.h:	../lib/libtiff.a
-	@chmod u+w ../common/tiff.h ../common/tiffio.h
-	@touch ../common/tiff.h ../common/tiffio.h
-
-normtiff:	normtiff.o ../lib/libtiff.a
+normtiff:	normtiff.o
 	$(CC) $(CFLAGS) -o normtiff normtiff.o -lrayt -ltiff $(MLIB)
 
-../lib/libtiff.a:	tiff/config.local
-	cd tiff ; ./configure -quiet -noninteractive -with-CC=$(CC) \
-		"-with-ENVOPTS=$(OPT)" ; \
-	cd libtiff ; make install
-
 ra_pixar:	ra_pixar.o ra_pixar.c
 	$(CC) $(CFLAGS) $(PIXAR_INCLUDE) -o $@ ra_pixar.o -lrayt \
 $(PIXAR_LIB)
@@ -246,8 +237,6 @@
 
 ra_t8.o:	../common/targa.h
 
-ra_tiff.o normtiff.o:	../common/tiff.h ../common/tiffio.h
-
 aedimage.o ximage.o ra_t8.o ra_pr.o:	pic.h
 
 ciq.o cut.o closest.o biq.o:	pic.h ciq.h
